climada/util/coordinates.py

```diff
@@ -1687,6 +1697,86 @@ def _ensure_utf8(val):
     return admin1_info, admin1_shapes
 
 
+def global_bounding_box():
+    """
+    Return global bounds in EPSG 4326
+
+    Returns
+    -------
+    tuple:
+        The global bounding box as (min_lon, min_lat, max_lon, max_lat)
+    """
+    return (-180, -90, 180, 90)
+
+
+def get_country_bounding_box(country_names, buffer=1.0):
+    """
+    Return bounding box in EPSG 4326 containing given countries.
+
+    Parameters
+    ----------
+    country_names : list or str
+        list with ISO 3166 alpha-3 codes of countries, e.g ['ZWE', 'GBR', 'VNM', 'UZB']
+    buffer : float, optional
+        Buffer to add to both sides of the bounding box. Default: 1.0.
+
+    Returns
+    -------
+    tuple
+        The bounding box containing all given coutries as (min_lon, min_lat, max_lon, max_lat)
+    """
+
+    country_geometry = get_country_geometries(country_names).geometry
+    longitudes, latitudes = [], []
+    for multipolygon in country_geometry:
+        for polygon in multipolygon.geoms:  # Loop through each polygon
+            for coord in polygon.exterior.coords:  # Extract exterior coordinates
+                longitudes.append(coord[0])
+                latitudes.append(coord[1])
+
+    return latlon_bounds(np.array(latitudes), np.array(longitudes), buffer=buffer)
+
+
+def bounds_from_cardinal_bounds(*, northern, eastern, western, southern):
+    """
+    Return and normalize bounding box in EPSG 4326 from given cardinal bounds.
+
+    Parameters
+    ----------
+    northern : (int, float)
+        Northern boundary of bounding box
+    eastern : (int, float)
+        Eastern boundary of bounding box
+    western : (int, float)
+        Western boundary of bounding box
+    southern : (int, float)
+        Southern boundary of bounding box
+
+    Returns
+    -------
+    tuple
+        The resulting normalized bounding box (min_lon, min_lat, max_lon, max_lat) with -180 <= min_lon < max_lon < 540
+
+    """
+
+    # latitude bounds check
+    if not ((90 >= northern > southern >= -90)):
+        raise ValueError(
+            "Given northern bound is below given southern bound or out of bounds"
+        )
+
+    if not (360 >= eastern >= -180) or not (360 >= western >= -180):
+        raise ValueError("Given eastern/western bounds are out of range (-180, 360).")
+    # order eastern and western coordinates
+    if western > eastern:
+        eastern += 360
+    if eastern > 360 and western > 180:
+        eastern -= 360
+        western -= 360
```

Comment:
```suggestion
    eastern = (eastern + 180) % 360 - 180
    western = (western + 180) % 360 - 180

    # Ensure eastern > western
    if western > eastern:
        eastern += 360
```

Reply:
I am pretty sure this works as well and avoid some condition checking.

Edit: I realise this includes bounding box outside of the (-180,360). You can keep the initial `if`, if you don't want that.